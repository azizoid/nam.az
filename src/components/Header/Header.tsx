import { useState } from 'react'

import Image from 'next/image'
import Link from 'next/link'
import { FaSearchLocation } from 'react-icons/fa'

import { MapModal } from '@/components'

export const Header = () => {
  const [showModal, setShowModal] = useState(false)

  return (
    <div className="bg-gray-100 py-2 px-2">
      <nav className="flex justify-between container mx-auto">
        <Link
          className="py-2 flex items-center content-start text-gray-500 hover:opacity-75"
          href="/"
        >
          <Image
            src="/favicon.png"
            width="30"
            height="30"
            alt="nam.az"
            className="mr-2"
          />
          Nam.az
        </Link>

        <div className="flex flex-col">
          <button
            className="bg-green-500 text-white active:bg-green-600 text-sm py-2 rounded-md shadow hover:shadow-lg outline-none focus:outline-none mr-1 ease-linear transition-all duration-150"
            type="button"
            onClick={() => setShowModal(true)}
            style={{ height: 36 }}
          >
            <div className="flex justify-center items-center">
              Xəritə &nbsp;
              <FaSearchLocation size="1em" />
            </div>
          </button>

          <MapModal
            open={showModal}
            onClose={() => setShowModal(false)}
          />

          <small>
            Bakı, Gəncə, <u>Şuşa</u> və digər
          </small>
        </div>
      </nav >
    </div >
  )
}
