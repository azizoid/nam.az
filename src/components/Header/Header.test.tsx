import { act, fireEvent, render, screen } from '@testing-library/react';
import { Header } from './Header'; // Replace with the actual import

// Mock the dynamically imported MapModal component
jest.mock('@/components/MapModal/MapModal', 
  () =>  ({ open }:{ open: boolean }) => (
      <div data-testid="mockMapModal">
        {open ? 'MapModal is open' : 'MapModal is closed'}
      </div>
    )
  )

describe('Header component', () => {
  it('should not render mock MapModal initially', async () => {
    await act(async () => {
      render(<Header />);
    });
    const mapModal = screen.queryByTestId('mockMapModal');
    expect(mapModal).toHaveTextContent('MapModal is closed');
  });

  it('should render mock MapModal when button is clicked', async () => {
    await act(async () => {
      render(<Header />);
    });
    
    const button = screen.getByText(/Xəritə/);
    fireEvent.click(button);
    
    const mapModal = screen.queryByTestId('mockMapModal');
    expect(mapModal).toHaveTextContent('MapModal is open');
  });
});
