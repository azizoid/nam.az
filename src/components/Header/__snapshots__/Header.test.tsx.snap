// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PrayerList component 1`] = `
<div>
  <nav
    class="flex justify-between container mx-auto"
  >
    <a
      class="py-2 flex items-center content-start text-gray-500 hover:opacity-75"
      href="/"
    >
      <img
        alt="nam.az"
        class="mr-2"
        height="30"
        src="/favicon.png"
        width="30"
      />
      Nam.az
    </a>
    <div
      class="flex flex-col"
    >
      <select
        aria-label="Haradasınız?"
        class="border-green-400 bg-green-50 text-sm rounded-md"
      >
        <option
          value="0"
        >
          Haradasınız?
        </option>
        <option
          value="1"
        >
          Bakı
        </option>
        <option
          value="2"
        >
          Sumqayıt
        </option>
        <option
          value="3"
        >
          Gəncə
        </option>
        <option
          value="4"
        >
          Lənkaran
        </option>
        <option
          value="5"
        >
          Mingəçevir
        </option>
        <option
          value="6"
        >
          Naxçıvan
        </option>
        <option
          value="7"
        >
          Ağdam
        </option>
        <option
          value="8"
        >
          Astara
        </option>
        <option
          value="9"
        >
          Beyləqan
        </option>
        <option
          value="10"
        >
          Cəlilabad
        </option>
        <option
          value="11"
        >
          Göyçay
        </option>
        <option
          value="12"
        >
          Qəbələ
        </option>
        <option
          value="13"
        >
          Qazax
        </option>
        <option
          value="14"
        >
          Quba
        </option>
        <option
          value="15"
        >
          Qusar
        </option>
        <option
          value="16"
        >
          Saatlı
        </option>
        <option
          value="17"
        >
          Sabirabad
        </option>
        <option
          value="18"
        >
          Şamaxı
        </option>
        <option
          value="19"
        >
          Şəki
        </option>
        <option
          value="20"
        >
          Şirvan
        </option>
        <option
          value="21"
        >
          Xaçmaz
        </option>
        <option
          value="22"
        >
          Yevlax
        </option>
        <option
          value="23"
        >
          Zaqatala
        </option>
        <option
          value="24"
        >
          Şuşa
        </option>
        <option
          value="25"
        >
          Marneuli
        </option>
        <option
          value="26"
        >
          Bolnisi
        </option>
      </select>
      <small>
        Bakı, Gəncə, 
        <u>
          Şuşa
        </u>
         və digər
      </small>
    </div>
  </nav>
</div>
`;
